assets/js/googlesitekit/datastore/site/info.js

```diff
@@ -0,0 +1,302 @@
+			yield actions.receiveSiteInfo( siteInfo );
+
+			if ( global._googlesitekitSiteData ) {
+				delete global._googlesitekitSiteData;
+			}
+
+			return;
+		}
+
+		global.console.error( 'Could not load core/site info.' );
+	},
+};
+
+export const selectors = {
+	/**
+	 * Gets all site info from this data store.
+	 *
+	 * Not intended to be used publicly; this is largely here so other selectors can
+	 * request data using the selector/resolver pattern.
+	 *
+	 * @since n.e.x.t
+	 * @private
+	 *
+	 * @param {Object} state Data store's state.
+	 * @return {Object} Site connection info.
+	 */
+	getSiteInfo( state ) {
+		const {
+			adminURL,
+			ampMode,
+			currentReferenceURL,
+			currentEntityID,
+			currentEntityTitle,
+			currentEntityType,
+			homeURL,
+			referenceSiteURL,
+		} = state.siteInfo || {};
+
+		return {
+			adminURL,
+			ampMode,
+			currentReferenceURL,
+			currentEntityID,
+			currentEntityTitle,
+			currentEntityType,
+			homeURL,
+			referenceSiteURL,
+		};
+	},
+
+	/**
+	 * Gets a site's admin URL.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {Object} state Data store's state.
+	 * @return {string|undefined} This site's admin URL.
+	 */
+	getAdminURL: createRegistrySelector( ( select ) => () => {
+		const { adminURL } = select( STORE_NAME ).getSiteInfo();
+
+		return adminURL;
+	} ),
+
+	/**
+	 * Gets a site's AMP mode.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {Object} state Data store's state.
+	 * @return {string|undefined} AMP Mode.
+	 */
+	getAMPMode: createRegistrySelector( ( select ) => () => {
+		const { ampMode } = select( STORE_NAME ).getSiteInfo();
+
+		return ampMode;
+	} ),
+
+	/**
+	 * Gets the current entity's ID.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {Object} state Data store's state.
+	 * @return {number|undefined} Current entity's ID.
+	 */
+	getCurrentEntityID: createRegistrySelector( ( select ) => () => {
+		const { currentEntityID } = select( STORE_NAME ).getSiteInfo();
+
+		return currentEntityID;
+	} ),
+
+	/**
+	 * Gets the current entity's title.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {Object} state Data store's state.
+	 * @return {string|undefined} Current entity's title.
+	 */
+	getCurrentEntityTitle: createRegistrySelector( ( select ) => () => {
+		const { currentEntityTitle } = select( STORE_NAME ).getSiteInfo();
+
+		return currentEntityTitle;
+	} ),
+
+	/**
+	 * Gets the current entity's title.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {Object} state Data store's state.
+	 * @return {string|undefined} Current entity's type.
+	 */
+	getCurrentEntityType: createRegistrySelector( ( select ) => () => {
+		const { currentEntityType } = select( STORE_NAME ).getSiteInfo();
+
+		return currentEntityType;
+	} ),
+
+	/**
+	 * Gets the current entity's reference URL.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {Object} state Data store's state.
+	 * @return {string|undefined} Current entity's reference URL.
+	 */
+	getCurrentReferenceURL: createRegistrySelector( ( select ) => () => {
```

Comment:
This should also become `getCurrentEntityURL`. This was not in the ACs like that, but after thinking further, it makes more sense to follow that naming pattern than a name based on existing PHP methods just because.